plugins/xsts/hu.bme.mit.gamma.theta.verification/src/hu/bme/mit/gamma/theta/verification/ThetaTraceGenerator.xtend

```diff
@@ -0,0 +1,135 @@
+package hu.bme.mit.gamma.theta.verification
+
+import java.util.List
+import hu.bme.mit.gamma.trace.model.ExecutionTrace
+import java.util.ArrayList
+import java.io.File
+import java.util.Scanner
+import java.util.logging.Level
+import hu.bme.mit.gamma.statechart.interface_.Package
+import java.util.logging.Logger
+import hu.bme.mit.gamma.verification.result.ThreeStateBoolean
+import hu.bme.mit.gamma.util.GammaEcoreUtil
+import hu.bme.mit.gamma.transformation.util.GammaFileNamer
+import org.eclipse.emf.common.util.EList
+import java.io.BufferedWriter
+import java.io.FileWriter
+import hu.bme.mit.gamma.util.FileUtil
+
+class ThetaTraceGenerator {
+	protected final GammaEcoreUtil ecoreUtil = GammaEcoreUtil.INSTANCE	
+	protected final extension GammaFileNamer fileNamer = GammaFileNamer.INSTANCE
+	
+	protected volatile boolean isCancelled
+	protected Process process
+	protected ThreeStateBoolean result	
+
+	final String ENVIRONMENT_VARIABLE_FOR_THETA_JAR = "THETA_XSTS_CLI_PATH"
+	protected final Logger logger = Logger.getLogger("GammaLogger")
+	
+	def List<ExecutionTrace> execute(File modelFile, boolean fullTraces, List<String> variableList, boolean noTransitionCoverage, boolean useAbstraction) {
+		val packageFileName = modelFile.name.unfoldedPackageFileName
+		val gammaPackage = ecoreUtil.normalLoad(modelFile.parent, packageFileName)
+		
+		return generateTraces(gammaPackage, modelFile, fullTraces, variableList, noTransitionCoverage, useAbstraction)
+	}
+	
+	private def List<ExecutionTrace> generateTraces(Object traceability, File modelFile, boolean fullTraces, List<String> variableList, boolean noTransitionCoverage, boolean useAbstraction) {
+		val traceDir = new File(modelFile.parent + File.separator + "traces")
+		cleanFolder(traceDir)		
+		val jar = System.getenv(ENVIRONMENT_VARIABLE_FOR_THETA_JAR)
+		var command = #["java", "-jar", jar] + #["--stacktrace", "--tracegen", "--search", "DFS", "--model", modelFile.canonicalPath, "--property", modelFile.canonicalPath]
```

Comment:
Add comments to highlight which arguments are essential for this kind of functionalitiy, e.g., `--tracegen`